src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -0,0 +1,232 @@
+import com.google.api.codegen.viewmodel.SimpleInitValueView;
+import com.google.api.codegen.viewmodel.StructureInitCodeLineView;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.TypeRef;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * InitCodeTransformer generates initialization code for a given method and then transforms
+ * it to a view object which can be rendered by a template engine.
+ */
+public class InitCodeTransformer {
+
+  public InitCodeView generateInitCode(MethodTransformerContext context, Iterable<Field> fields) {
+    Map<String, Object> initFieldStructure = createInitFieldStructure(context);
+    InitCodeGenerator generator = new InitCodeGenerator();
+    InitCode initCode =
+        generator.generateRequestFieldInitCode(context.getMethod(), initFieldStructure, fields);
+
+    return InitCodeView.newBuilder()
+        .lines(generateSurfaceInitCodeLines(context, initCode))
+        .fieldSettings(getFieldSettings(context, initCode.getArgFields()))
+        .build();
+  }
+
+  public InitCodeView generateRequestObjectInitCode(MethodTransformerContext context) {
+    Map<String, Object> initFieldStructure = createInitFieldStructure(context);
+    InitCodeGenerator generator = new InitCodeGenerator();
+    InitCode initCode =
+        generator.generateRequestObjectInitCode(context.getMethod(), initFieldStructure);
+
+    return InitCodeView.newBuilder()
+        .lines(generateSurfaceInitCodeLines(context, initCode))
+        .fieldSettings(getFieldSettings(context, initCode.getArgFields()))
+        .build();
+  }
+
+  public Map<String, Object> createInitFieldStructure(MethodTransformerContext context) {
+    Map<String, String> fieldNamePatterns = context.getMethodConfig().getFieldNamePatterns();
+
+    ImmutableMap.Builder<String, InitValueConfig> initValueConfigMap = ImmutableMap.builder();
+    for (Map.Entry<String, String> fieldNamePattern : fieldNamePatterns.entrySet()) {
+      CollectionConfig collectionConfig = context.getCollectionConfig(fieldNamePattern.getValue());
+      String apiWrapperClassName =
+          context.getNamer().getApiWrapperClassName(context.getInterface());
+      InitValueConfig initValueConfig =
+          InitValueConfig.create(apiWrapperClassName, collectionConfig);
+      initValueConfigMap.put(fieldNamePattern.getKey(), initValueConfig);
+    }
+    Map<String, Object> initFieldStructure =
+        FieldStructureParser.parseFields(
+            context.getMethodConfig().getSampleCodeInitFields(), initValueConfigMap.build());
+    return initFieldStructure;
+  }
+
+  public List<InitCodeLineView> generateSurfaceInitCodeLines(
+      MethodTransformerContext context, InitCode initCode) {
+    List<InitCodeLineView> surfaceLines = new ArrayList<>();
+    for (InitCodeLine line : initCode.getLines()) {
+      switch (line.getLineType()) {
```

Comment:
the line type enum is used in both the set of classes from init code and the views that represent them; it is more of a conceptual level than the exact implementing class.